lib/sht4x/transport.ex

```diff
@@ -5,16 +5,26 @@ defmodule SHT4X.Transport do
 
   typedstruct do
     field(:address, 0..127, enforce: true)
-    field(:ref, reference, enforce: true)
+    field(:ref, any, enforce: true)
     field(:retries, non_neg_integer, default: 0)
     field(:read_fn, (pos_integer -> {:ok, binary} | {:error, any}), enforce: true)
     field(:write_fn, (iodata -> :ok | {:error, any}), enforce: true)
     field(:write_read_fn, (iodata, pos_integer -> {:ok, binary} | {:error, any}), enforce: true)
   end
 
-  @spec new(reference, 0..127, non_neg_integer) :: {:ok, t()} | {:error, any}
-  def new(ref, address, retries)
-      when is_reference(ref) and is_integer(address) and is_integer(retries) do
+  @spec new(any, 0..127, non_neg_integer) :: {:ok, t()} | {:error, any}
+  def new(bus_name, address, retries) when is_binary(bus_name) do
+    case Circuits.I2C.open(bus_name) do
+      {:error, err} ->
+        {:error, err}
+
+      {:ok, ref} ->
+        new(ref, address, retries)
+    end
+  end
+
+  # We want to allow ref to be of any type so that we can use sim etc as needed.
+  def new(ref, address, retries) when is_integer(address) and is_integer(retries) do
```

Comment:
Is this version of `new` actually called externally? I think that it could be made into a private function and then the spec could be the following for both Circuits.I2C v1 and v2:

```elixir
@spec new(String.t(), Circuits.I2C.address(), non_neg_integer) :: {:ok, t()} | {:error, any}
```